arangod/Aql/OptimizerRules.cpp

```diff
@@ -6879,6 +6879,49 @@ void arangodb::aql::geoIndexRule(Optimizer* opt, std::unique_ptr<ExecutionPlan>
   opt->addPlan(std::move(plan), rule, mod);
 }
 
+static bool isInnerPassthroughNode(ExecutionNode* node) {
+  switch (node->getType()) {
+    case ExecutionNode::CALCULATION:
+    case ExecutionNode::SUBQUERY:
+      return true;
+    case ExecutionNode::SINGLETON:
+    case ExecutionNode::ENUMERATE_COLLECTION:
+    case ExecutionNode::ENUMERATE_LIST:
+    case ExecutionNode::FILTER:
+    case ExecutionNode::LIMIT:
+    case ExecutionNode::SORT:
+    case ExecutionNode::COLLECT:
+    case ExecutionNode::INSERT:
+    case ExecutionNode::REMOVE:
+    case ExecutionNode::REPLACE:
+    case ExecutionNode::UPDATE:
+    case ExecutionNode::NORESULTS:
+    case ExecutionNode::UPSERT:
+    case ExecutionNode::TRAVERSAL:
+    case ExecutionNode::INDEX:
+    case ExecutionNode::SHORTEST_PATH:
+    case ExecutionNode::K_SHORTEST_PATHS:
+    case ExecutionNode::ENUMERATE_IRESEARCH_VIEW:
+    case ExecutionNode::RETURN:
+      return false;
+    case ExecutionNode::REMOTE:
+    case ExecutionNode::DISTRIBUTE:
+    case ExecutionNode::SCATTER:
+    case ExecutionNode::GATHER:
+    case ExecutionNode::REMOTESINGLE:
+      THROW_ARANGO_EXCEPTION_MESSAGE(
```

Comment:
I think it is possible that a REMOTESINGLE operation is found here, because the optimizer rule that inserts this type of node runs directly before the sort-limit rule.
So we should exchange the order of the two rules.

Reply:
Like so? https://github.com/arangodb/arangodb/pull/10014/commits/2db9978fde51925406b9af26eb365fc5829aed77